src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,47 @@
+.calendar {
+  margin: 71.5px 149px;
+  font-size: 30px;
+  padding: $square-paddings;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $square-spacing;
+  width: $calendar-width;
+
+  &__day {
+    @include square($square-size);
+
+    background-color: $background-color;
+    text-align: center;
+    line-height: $square-size;
+    cursor: pointer;
+    border: 1px solid black;
+    transition:
+      transform 500ms,
+      background-color 500ms;
+
+    &:hover {
+      background-color: $background-color-hover;
+      transform: translateY(-20px);
+    }
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: '#{$i}';
+      }
+    }
+  }
+
+  &--month-length {
+    @for $i from 28 through 31 {
+      &--#{$i} :nth-child(n + #{$i + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  @each $day, $count in $start-days {
```

Comment:
The variable `$start-days` is used here but needs to be defined in your SCSS files to avoid compilation errors.